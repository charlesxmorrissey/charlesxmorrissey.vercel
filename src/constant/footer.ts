import { JSXElementConstructor, SVGProps } from 'react'

import EmailIcon from 'images/email.svg'
import GithubIcon from 'images/github.svg'
import LinkedInIcon from 'images/linkedin.svg'

export interface LinkData {
  Icon: JSXElementConstructor<SVGProps<SVGElement>>
  link: string
  name: string
  options?: {
    rel?: string
    target?: string
  }
}

export const footerData: LinkData[] = [
  {
    Icon: EmailIcon,
    link: 'mailto:hi@charles-x.com?subject=hello%20from%20website',
    name: 'Email',
    options: {
      rel: 'noreferrer',
      target: '_blank',
    },
  },
  {
    Icon: GithubIcon,
    link: 'https://github.com/charlesxmorrissey',
    name: 'Github',
  },
  {
    Icon: LinkedInIcon,
    link: 'https://www.linkedin.com/in/charles-x-morrissey-b366976',
    name: 'LinkedIn',
  },
]
