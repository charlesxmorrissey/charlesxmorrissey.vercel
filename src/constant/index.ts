import { JSXElementConstructor, SVGProps } from 'react'

import EmailIcon from 'images/email.svg'
import GithubIcon from 'images/github.svg'
import LinkedInIcon from 'images/linkedin.svg'

interface AppData {
  description: string
  name: string
  title: string
}

interface LinkDataAttributes {
  rel?: string
  target?: string
}

export interface LinkData {
  Icon: JSXElementConstructor<SVGProps<SVGElement>>
  link: string
  name: string
  options?: LinkDataAttributes
}

export const appData: AppData = {
  description:
    'I build apps for the web using React, Vue, Node, ES6, Webpack, and more.',
  name: 'Charles X. Morrissey',
  title: 'Senior Frontend Engineer',
}

export const socialData: LinkData[] = [
  {
    Icon: EmailIcon,
    link: `mailto:hi@charles-x.com?subject=${encodeURIComponent(
      'hello from website'
    )}`,
    name: 'Email',
    options: {
      rel: 'noreferrer',
      target: '_blank',
    },
  },
  {
    Icon: GithubIcon,
    link: 'https://github.com/charlesxmorrissey',
    name: 'Github',
  },
  {
    Icon: LinkedInIcon,
    link: 'https://www.linkedin.com/in/charles-x-morrissey-b366976',
    name: 'LinkedIn',
  },
]
