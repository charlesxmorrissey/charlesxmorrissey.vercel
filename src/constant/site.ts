export const siteData = {
  description:
    'I build apps for the web using React, Vue, Node, ES6, Webpack, and more.',
  name: 'Charles X. Morrissey',
  title: 'Senior Frontend Engineer',
}
