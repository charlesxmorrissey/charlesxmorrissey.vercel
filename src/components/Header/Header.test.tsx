import { render, screen } from '@testing-library/react'
import { APP_DATA } from 'constant'

import { Header } from './'

const { description, name } = APP_DATA
const { getByRole, getByText } = screen

describe('Header', () => {
  it('should render the header with required props', () => {
    render(<Header {...APP_DATA} />)

    expect(getByRole('heading', { level: 1 })).toBeInTheDocument()
    expect(getByRole('heading', { level: 1 })).toHaveTextContent(name)
    expect(getByText(description)).toBeInTheDocument()
  })

  it('renders exactly one level-1 heading with the correct accessible name', () => {
    render(<Header {...APP_DATA} />)

    const h1 = screen.getAllByRole('heading', { level: 1 })

    expect(h1).toHaveLength(1)
    expect(h1[0]).toHaveAccessibleName(name)
    expect(h1[0]).toBeVisible()
  })

  it('renders the description as visible text', () => {
    render(<Header {...APP_DATA} />)

    const desc = screen.getByText(description)

    expect(desc).toBeVisible()
    expect(desc.tagName.toLowerCase()).not.toBe('h1')
  })

  it('matches the rendered snapshot', () => {
    const { container } = render(<Header {...APP_DATA} />)

    expect(container).toMatchSnapshot()
  })
})
