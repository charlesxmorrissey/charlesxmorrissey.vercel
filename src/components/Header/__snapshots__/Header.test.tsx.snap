// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > matches the rendered snapshot 1`] = `
<div>
  <header
    class="_header_a5b977"
  >
    <h1
      class="_headerTitle_a5b977"
    >
      Charles X. Morrissey
    </h1>
    <p
      class="_headerText_a5b977"
    >
      I'm passionate about building user-facing web products and overcoming technical hurdles to ensure their success.
    </p>
  </header>
</div>
`;
